### PR TITLE
Phinx recipe

### PR DESCRIPTION
**Pull request by ket4yii**
# Phinx recipe
### Installing

``` php
// deploy.php

require 'vendor/deployphp/recipes/recipes/phinx.php';
```
### Configuration options

All options are in the environment variable `phinx` specified as a dictionary
(instead of the `phinx_path` variable).
All parameters are _optional_, but you can specify them with a dictionary(to change all parameters)
or by deployer dot notation(to change one option).
#### Phinx environment variable
- `phinx.environment`
- `phinx.date`
- `phinx.configuration` N.B. current directory is the project directory
- `phinx.target`
- `phinx.seed`
- `phinx.parser`
#### Phinx path environment variable
- `phinx_path` Specify phinx path(by default phinx is searched in 
  $PATH, ./vendor/bin and ~/.composer/vendor/bin)
#### Example of usage

``` php
//deploy.php

env('phinx_path', '/usr/local/phinx/bin/phinx');
env('phinx.environment', 'production');
env('phinx.configuration', './migration/.phinx.yml');

after('cleanup', 'phinx:migrate');

//Or set it for a specific server
server('dev', 'my-dev-server.local')
    ->user('user')
    ->env('deploy_path', '/var/www')
    ->env('phinx.environment', 'development')
    ->env('phinx_path', '');
```
### Tasks
- `phinx:migrate` Migrate your database
- `phinx:rollback` Rollback your database
- `phinx:seed` Run seeds for your database
### Suggested Usage

You can run all tasks after or before any 
tasks(but you need to specify external configs for phinx).
If you use internal configs(which are in your project) you need 
to run it after `deploy:update_code` task is completed.
### Read more

For further reading check the [phinx.org](https://phinx.org). Options description are on the [commands page](http://docs.phinx.org/en/latest/commands.html).

### CODE REVIEW

**Comment by oanhnn** [comment]
Thank you!
We will test and merge. Please rebase to less commit :smile:

**Comment by ket4yii** [comment]
I hope recipe will be useful:smile:. All commits were squashed.

**Comment by ket4yii** [comment]
Are there any news about the checking and merging?

File: recipes/phinx.php [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    'phinx', [
+        'environment' => '',
+        'date' => '',
+        'configuration' => '', 
+        'target' => '',
+        'seed' => '',
+        'parser' => ''
+    ]
+);
+env('phinx_path', 'getPhinx');
```
Comment (mbrodala) [inline recipes/phinx.php]: Can you move this function to an inline closure to not pollute the global namespace?

**Comment by ket4yii** [comment]
Done.

**Comment by mbrodala** [comment]
@ket4yii Could you check if you can get rid of the global functions and convert them to closures where possible?

**Comment by ket4yii** [comment]
Like this?

**Comment by mbrodala** [comment]
I'm sure it works this way but I don't know if `env()` should be used like this. @oanhnn What do you think?

**Comment by ket4yii** [comment]
But, I think, it's used in the same way [here](https://github.com/deployphp/recipes/blob/master/recipes/local.php).

**Comment by elfet** [comment]
Please, rebase your PR.

**Comment by ket4yii** [comment]
Done

**Comment by elfet** [comment]
Cant merge it. Can you create another PR?

**Pull request closed without merging**